src/software/ai/hl/stp/play/shoot_or_pass/shoot_or_pass_play_fsm.h

```diff
@@ -0,0 +1,147 @@
+
+    struct ControlParams
+    {
+    };
+
+    DEFINE_PLAY_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+
+    /**
+     * Creates a shoot or pass play fsm
+     *
+     * @param play_config the play config for this play fsm
+     */
+    explicit ShootOrPassPlayFSM(std::shared_ptr<const PlayConfig> play_config);
+
+    /**
+     * Updates the offensive positioning tactics
+     *
+     *
+     * @param the ranked zones to look for offensive positions in
+     * @param pass_eval The pass evaluation to help find best passes
+     * @param num_tactics the number of tactics to return
+     */
+    void updateOffensivePositioningTactics(std::vector<EighteenZoneId> ranked_zones,
+                                           PassEvaluation<EighteenZoneId> pass_eval,
+                                           unsigned int num_tactics);
+
+    /**
+     * Action that looks for a pass
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     */
+    void lookForPass(const Update& event);
+
+    /**
+     * Action to restart looking for a pass
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     */
+    void startLookingForPass(const Update& event);
+
+    /**
+     * Action to take a pass
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     */
+    void takePass(const Update& event);
+
+    /**
+     * Guard to check if a pass has been found
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     *
+     * @return whether a pass has been found
+     */
+    bool passFound(const Update& event);
+
+    /**
+     * Guard on whether to abort the pass
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     *
+     * @return whether the pass should be aborted
+     */
+    bool shouldAbortPass(const Update& event);
+
+    /**
+     * Guard on whether the pass has completed
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     *
+     * @return whether the pass has completed
+     */
+    bool passCompleted(const Update& event);
+
+    /**
+     * Guard on whether a shot has been taken
+     *
+     * @param event the ShootOrPassPlayFSM Update event
+     *
+     * @return whether the shot has been taken
+     */
+    bool tookShot(const Update& event);
+
+    auto operator()()
+    {
+        using namespace boost::sml;
+
+        DEFINE_SML_STATE(AttemptShotState)
+        DEFINE_SML_STATE(TakePassState)
+        DEFINE_SML_STATE(StartState)
+        DEFINE_SML_EVENT(Update)
+
+        DEFINE_SML_ACTION(lookForPass)
+        DEFINE_SML_ACTION(startLookingForPass)
+        DEFINE_SML_ACTION(takePass)
+
+        DEFINE_SML_GUARD(passFound)
+        DEFINE_SML_GUARD(shouldAbortPass)
+        DEFINE_SML_GUARD(passCompleted)
+        DEFINE_SML_GUARD(tookShot)
+
+        return make_transition_table(
+            // src_state + event [guard] / action = dest_state
+            *StartState_S + Update_E / startLookingForPass_A        = AttemptShotState_S,
```

Comment:
startLookingForPass and lookForPass seem pretty similar, can we combine it or is there a reason it's separate?

Reply:
it needs to reset the Attacker and Receiver Tactics when it starts looking for a new pass